csharp/device/Microsoft.Azure.Devices.Client/Common/SHA.cs

```diff
@@ -0,0 +1,211 @@
+// 
+//  Code by Elze Kool
```

Comment:
If you know him already, personally or online, I'd suggest you do it but if you don't, it doesn't matter if it's you or me. I'll be happy to do it. just let me know :)